src/main/java/com/coveros/selenified/element/check/azzert/AssertState.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2019 Coveros, Inc.
+ *
+ * This file is part of Selenified.
+ *
+ * Selenified is licensed under the Apache License, Version
+ * 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy
+ * of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.coveros.selenified.element.check.azzert;
+
+import com.coveros.selenified.OutputFile;
+import com.coveros.selenified.element.Element;
+import com.coveros.selenified.element.check.State;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+/**
+ * State extends Asserts to provide some additional verification capabilities.
+ * It will handle all verifications performed on the actual element. These
+ * asserts are custom to the framework, and in addition to providing easy object
+ * oriented capabilities, they take screenshots with each verification to
+ * provide additional traceability, and assist in troubleshooting and debugging
+ * failing tests. State checks that elements are in a particular state.
+ *
+ * @author Max Saperstone
+ * @version 3.1.0
+ * @lastupdate 9/13/2018
+ */
+public class AssertState implements State {
+
+    // this will be the name of the file we write all commands out to
+    private final OutputFile file;
+
+    // this is the element that all actions will be performed on
+    private final Element element;
+
+    public AssertState(Element element, OutputFile file) {
+        this.element = element;
+        this.file = file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OutputFile getOutputFile() {
+        return file;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Element getElement() {
+        return element;
+    }
+
+    // ///////////////////////////////////////
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    /**
+     * Verifies that the element is present. If the element isn't present, it
+     * waits up to the default time (5 seconds) for the element, before marking
+     * this verification as a failure. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     */
+    public void present() {
+        assertTrue("Expected Element Not Present", checkPresent(0, 0));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Expected Element Not Present" 9 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1192)